lib/spack/spack/test/concretize.py

```diff
@@ -1138,14 +1137,46 @@ def test_external_package_versions(self, spec_str, is_external, expected):
         assert s.external == is_external
         assert s.satisfies(expected)
 
+    def test_reuse_does_not_overwrite_root_dev_specs(
+            self, tmpdir, mock_packages, install_mockery, mock_fetch):
+        # potential parameters
+        package_name = 'dev-build-test-install'
+        package_version = '0.0.0'
+        spec_str = package_name
+        dev_spec_str = '%s@%s dev_path=/fake' % (package_name,
+                                                 package_version)
+        # concretize and install a non-dev version
+        s = Spec(spec_str).concretized()
+        s.package.do_install(fake=True)
+        # concretize a dev version
+        dev_spec = Spec(dev_spec_str).concretized(reuse=True)
+        assert dev_spec.dag_hash() is not s.dag_hash()
```

Comment:
```suggestion
        assert dev_spec.dag_hash() != s.dag_hash()
```
`is not` checks whether they are the same string object in memory, `!=` checks equality.

Reply:
🤔 thanks!